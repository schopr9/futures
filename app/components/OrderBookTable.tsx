import React from 'react';
import { ScrollView, StyleSheet, View } from 'react-native';
import { DataTable, Text } from 'react-native-paper';
import { useSelector } from 'react-redux';

import { IOrderBookInitialState } from 'app/models/reducers/orderBook';

interface IState {
  orderBookReducer: IOrderBookInitialState;
}

const OrderBookTable: React.FC = () => {
  const asks = useSelector((state: IState) => state.orderBookReducer?.asks);
  const bids = useSelector((state: IState) => state.orderBookReducer?.bids);

  return (
    <>
      <DataTable>
        <DataTable.Header>
          <DataTable.Title sortDirection="descending">
            Price ($)
          </DataTable.Title>
          <DataTable.Title numeric>Size</DataTable.Title>
          <DataTable.Title numeric>Total</DataTable.Title>
        </DataTable.Header>
        <ScrollView contentContainerStyle={{ paddingBottom: 160 }}>
          <Text style={styles.text}>Asks</Text>
          {!!asks &&
            asks.map(({ price, size, total }) => {
              return (
                <DataTable.Row key={price}>
                  <DataTable.Cell numeric>{price}</DataTable.Cell>
                  <DataTable.Cell numeric>{size}</DataTable.Cell>
                  <DataTable.Cell numeric>{total}</DataTable.Cell>
                </DataTable.Row>
              );
            })}

          <Text style={styles.text}>Bids</Text>
          {!!bids &&
            bids.map(({ price, size, total }) => {
              return (
                <DataTable.Row key={price}>
                  <DataTable.Cell numeric>{price}</DataTable.Cell>
                  <DataTable.Cell numeric>{size}</DataTable.Cell>
                  <DataTable.Cell numeric>{total}</DataTable.Cell>
                </DataTable.Row>
              );
            })}
        </ScrollView>
      </DataTable>
    </>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    fontSize: 18,
    marginTop: 5,
  },
});

export default OrderBookTable;
