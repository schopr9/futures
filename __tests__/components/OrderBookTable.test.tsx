import React from 'react';
import { render } from '@testing-library/react-native';
import OrderBookTable from 'app/components/OrderBookTable';

import * as redux from 'react-redux';

describe('Order book table', () => {
  it('mounts Maintenance route if maintenance mode is enabled', () => {
    const useDispatchSpy = jest.spyOn(redux, 'useSelector');
    useDispatchSpy.mockReturnValue([{ price: 67.89, size: 20, total: 100 }]);
    const { toJSON } = render(<OrderBookTable />);
    expect(toJSON).toMatchSnapshot();
  });
});
