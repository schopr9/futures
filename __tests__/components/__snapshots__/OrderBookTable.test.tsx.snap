// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order book table mounts Maintenance route if maintenance mode is enabled 1`] = `[Function]`;
